go/rpcclientlib/obscuro_client.go

```diff
@@ -49,11 +49,11 @@ type Client interface {
 	Stop()
 
 	// SetViewingKey sets the current viewing key on the client to be used for all sensitive request decryption
-	SetViewingKey(viewingKey *ecies.PrivateKey, pubKeyBytes []byte)
+	SetViewingKey(viewingKeyAddr common.Address, viewingKey *ecies.PrivateKey)
 
 	// RegisterViewingKey takes a signature for the public key, it verifies the signed public key matches the currently set private viewing key
 	//	and then submits it to the enclave
-	RegisterViewingKey(signerAddr common.Address, signature []byte) error
+	RegisterViewingKey(signature []byte, viewingPubKeyBytes []byte) error
```

Comment:
@BedrockSquirrel It can just be added as a todo for now, but it seems wrong to me that we re-pass in the viewingPubKeyBytes here, when we've already set the viewing key, so we know what the viewing key is.

Looking at it now, it seems to me that it would be cleaner to get rid of the `SetViewingKey` method, and store the generated viewing key on the wallet side in the wallet extension until it can be signed by MetaMask and submitted to the enclave.

Thoughts?